share/eds/hash.go

```diff
@@ -0,0 +1,12 @@
+package eds
+
+import "fmt"
+
+type DataHash []byte
```

Comment:
We usually keep such types in the root `share` pkg, and `eds` right now does not hold any core type definitions but only relevant components, so suggesting to move it to `share`. However, the `eds` pkg grows so fast that it might become a root one on its own, but that's a different discussion after ITN refactorings. 

Also go doc.

Reply:
https://github.com/celestiaorg/celestia-node/pull/1439/commits/a3fa93bc0a92fc01141c6fe800e5c39958a34451